index.js

```diff
@@ -25,3 +25,20 @@ const port = process.env.port || 3000;
 app.listen(port,()=>console.log(`listening on port ${port}....`));
 // functions
 
+//post function
+app.post('/api/courses',(req,res)=>{
+    const schema = Joi.object({
+        name: Joi.string().min(3).required()
+    });
+    
+        const result = schema.validate(req.body);
+  if(result.error){res.status(400).send(result.error.details[0].message);
+    return;
+}
+const course = {
+    id: courses.length +1,
+    name : req.body.name
+};
+courses.push(course);
+res.send(course);
+});
```

Comment:
The handler shouldn't return it only uses `res` to send response.

MIND THE INDENTATION, the `}` in the middle, I thought that was the closing of you handler, and was just about to give you 0.